### DISCUSSION TITLE
Is there a way to extend NPM package hosted on private Artifactory?

### DISCUSSION THREAD

**Discussion by yli-dev** [stoplightio/spectral#1683] [Q&A]
Hi team, we've created a ruleset with custom functions, bundled it as NPM package and hosted it on our private Artifactory.

It seems all CLI, Stoplight studio and GitHub action are fetching the package through unpkg.com. Is there way to override the URL with our custom registry?

Thanks,
Yli

**Comment by P0lip** [comment]
Hey!
I'm delighted to inform you that Spectral v6 (which is just around the corner, by the way) will introduce a new format of the ruleset.
Instead of authoring JSON/YAML, you'll be able to specify a JS file, for instance:

```js
// .spectral.js
import myRuleset from 'my-npm-package';

export default {
  extends: myRuleset,
};
```

Note - it's going to be complementary to JSON/YAML, therefore the syntax of that rule will strictly follow JSON/YAML.

**Reply by yli-dev** [reply]
A nice feature to have! Is there documentation on how it can be used to facilitate us writing and using custom functions?

**Reply by P0lip** [reply]
> Is there documentation on how it can be used to facilitate us writing and using custom functions?

Not yet, but the idea is that the ruleset file wouldn't behave differently than any other Node.js module.
That means you could leverage npm or yarn, and the custom function will simply follow the Node.js module resolution mechanism.
Does that make sense?

**Reply by yli-dev** [reply]
I got it now. Thanks for clarifying 👍
